src/tests/ftest/util/server_utils.py

```diff
@@ -50,6 +58,429 @@ class ServerFailed(Exception):
+
+                # Read avocado test yaml file.
+                try:
+                    with open(avocado_yaml_file, 'r') as rfile:
+                        filedata = rfile.read()
+                    # Remove !mux for yaml load
+                    new_value_set = yaml.safe_load(
+                        filedata.replace('!mux', ''))
+                except Exception as err:
+                    self.log.info("<SERVER> Exception occurred: {0}".format(
+                        str(err)))
+                    traceback.print_exception(
+                        err.__class__, err, sys.exc_info()[2])
+                    raise ServerFailed(
+                        "Failed to Read {}".format(
+                            '{}.tmp'.format(avocado_yaml_file)))
+
+            # Update values from avocado_testcase.yaml in DAOS yaml variables.
+            if new_value_set:
+                if 'server' in new_value_set['server_config']:
+                    for key in new_value_set['server_config']['server']:
+                        self.data['servers'][0][key] = \
+                            new_value_set['server_config']['server'][key]
+                for key in new_value_set['server_config']:
+                    if 'server' not in key:
+                        self.data[key] = \
+                            new_value_set['server_config'][key]
+
+            # Check if nvme and scm are enabled
+            srv_cfg = self.data['servers'][0]
+            if 'bdev_class' in srv_cfg and srv_cfg['bdev_class'] == "nvme":
+                self.nvme = True
+            if 'scm_class' in srv_cfg and srv_cfg['scm_class'] == "dcpm":
+                self.scm = True
+
+            # if specific log file name specified use that
+            if test.server_log:
+                self.data['servers'][0]['log_file'] = test.server_log
+
+        def is_nvme(self):
+            """Return if NVMe is provided in the configuration."""
+            return self.nvme is not None
+
+        def is_scm(self):
+            """Return if SCM (ram) is provided in the configuration."""
+            return self.scm is not None
+
+        def create_yaml(self, yamlfile):
+            """Create the DAOS server config YAML file based on Avocado test
+                Yaml file.
+
+            Args:
+                yamlfile (str): full path and name of yaml file
+
+            Raises:
+                ServerFailed: if there is an reading/writing yaml files
+
+            Returns:
+                (str): Absolute path of create server yaml file
+
+            """
+            # Write self.data dictionary in to AVOCADO_FILE
+            # This will be used to start with daos_server -o option.
+            try:
+                with open(yamlfile, 'w') as wfile:
+                    yaml.dump(self.data, wfile, default_flow_style=False)
+            except Exception as err:
+                self.log.info("<SERVER> Exception occurred:{}".format(str(err)))
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments